app/src/main/java/com/javahelp/frontend/activity/ProviderRegistrationActivity.java

```diff
@@ -25,14 +50,109 @@ protected void onCreate(Bundle savedInstanceState) {
         binding.setPdata(providerRegistrationViewmodel);
         binding.setLifecycleOwner(this);
 
-        binding.signupbtn.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                Intent intent = new Intent(ProviderRegistrationActivity.this,
-                        FrontPage2Activity.class);
-                startActivity(intent);
-            }
-        });
+        providerRegistrationViewmodel.getRegisterResult().observe(this, this::updateOnRegisterResult);
+
+        binding.signupbtn.setOnClickListener(this::registerClick);
+
+    }
+
+    /**
+     * Called when the Register button is clicked
+     *
+     * @param v {@link View} that was clicked
+     */
+    private void registerClick(View v) {
+        providerRegistrationViewmodel.setUsername(binding.username.getText().toString());
+        providerRegistrationViewmodel.setPassword1(binding.password.getText().toString());
+        providerRegistrationViewmodel.setPassword2(binding.repassword.getText().toString());
+        providerRegistrationViewmodel.setProviderUserInfo(binding.email.getText().toString(),
+                binding.firstname.getText().toString(),
```

Comment:
This should be `binding.practicename.getText().toString()` and there should be (optional) one for `isCertified` for `Providers` (`false` by default)

Reply:
the practice name is set in the ViewModel by  "this.ProviderUserInfo.setPracticeName(firstname + lastname);"
Same as the certified which is set as false by "this.providerUserInfo.setCertified(false);"
you can have a look at ProviderRegistrationViewmodel

Reply:
Good idea!